### PR TITLE
bazel: add sample-only option to go_gapic_library

### LINKED ISSUE #233: bazel: add sample-only option to go_gapic_library
The `sample-only` plugin option disables generation of the GAPIC and only runs sample gen. This option should be reflected in the go_gapic_library macro

### PR DESCRIPTION
**Pull request by idoqo**
Adds the `sample` and `sample-only` protoc plugin options to the bazel rule and should close #233.

### CODE REVIEW

**Comment by noahdietz** [comment]
I didn't realize this but we need to add _another_ attribute for a plugin flag to enable the sample gen feature.

`gapic-config` is the name of the plugin option, but we should add an optional attribute to the Bazel rule named `gapic_yaml` to stay consistent with other GAPIC rules. It is another `file_args` item.

Essentially, add the following
```
gapic_yaml = None,
...
if gapic_yaml:
    file_args[gapic_yaml] = "gapic-config"
```

I have also updated the test repo, so pull that and build the `echo_go_gapic_sample` target to test! https://github.com/noahdietz/bazel-microgen-test/commit/adeaa41c06b210f9059e06a59b5aeda25e7e51ec

Also, make sure to pull recent changes to this repo to keep up with master :)

**Comment by idoqo** [comment]
I've added the gapic_yaml arg and updated the readme too. Sorry, I'm a bit confused about something though - when I generated the samples using protoc (for the bazel-microgen-test repo), I get a `showcase_echo_sample.go` file in `$OUTPUT_DIR/samples` but with bazel, it's just the `srcjar` files in bazel-bin. That means bazel isn't generating those samples right?

**Comment by noahdietz** [comment]
> it's just the srcjar files in bazel-bin

Actually, this jar contains all of the generated code! This is a packaging mechanism all of our languages follow. If you list the contents, you will the generated files `jar -tf bazel-bin/echo_go_gapic.srcjar` (or whatever the ja name is).

File: README.md [hunk 1]
```diff
@@ -147,6 +147,10 @@ following attributes:
 
   * `service_yaml`: a label for a service YAML file.
     * _Note: This option will eventually be deprecated._
+  * `sample`: a path to a sample config file. There 
+  can exist multiples of this.
+  * `gapic_yaml`: path to a gapic configuration file.
```
Comment (noahdietz) [inline README.md]: Can you add a sub-bullet under this one stating it's use is only for sample generation, please? Something like:

* `gapic_yaml`: path to a gapic configuration file.
  * _Note: This config is only used for sample generation._

File: go.mod [hunk 2]
```diff
@@ -12,7 +12,7 @@ require (
 	github.com/opennota/wd v0.0.0-20180911144301-b446539ab1e7 // indirect
 	github.com/russross/blackfriday v2.0.0+incompatible // indirect
 	github.com/shurcooL/sanitized_anchor_name v1.0.0 // indirect
-	google.golang.org/genproto v0.0.0-20200207204624-4f3edf09f4f6
+	google.golang.org/genproto v0.0.0-20200211111953-2dc5924e3898
```
Comment (noahdietz) [inline go.mod]: Can you revert the changes to the `go.mod` and `go.sum`, please? We have a bot that updates the deps.

**Comment by noahdietz** [comment]
Also Fixes #235. These should've been in the same issue.

**Comment by noahdietz** [comment]
Great job @idoqo ! Thank you so much for the contribution, we really appreciate it.
